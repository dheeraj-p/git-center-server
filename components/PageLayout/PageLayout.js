import { Fragment } from 'react';
import Head from 'next/head';
import GlobalStyles from '../../styles/global.style';
import Header from '../Header/Header';

export default function PageLayout({ title, children }) {
  return (
    <Fragment>
      <Head>
        <title>{title}</title>
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
        <link
          rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
          integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm"
          crossOrigin="anonymous"
        />
      </Head>
      <div>
        <Header />
        {children}
      </div>
      <style jsx>{GlobalStyles}</style>
    </Fragment>
  );
}
