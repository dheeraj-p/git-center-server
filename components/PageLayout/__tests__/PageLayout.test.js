import { shallow } from 'enzyme';
import PageLayout from '../PageLayout';

const DummyComponent = () => {
  return <div>Dummy</div>;
};

describe('PageLayout', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <PageLayout title="title">
        <DummyComponent />ÃŽ
      </PageLayout>
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
