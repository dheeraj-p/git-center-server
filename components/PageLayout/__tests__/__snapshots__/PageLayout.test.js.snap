// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout should render properly 1`] = `
"<Fragment>
  <Head>
    <title className=\\"jsx-235099215\\">
      title
    </title>
    <link rel=\\"stylesheet\\" href=\\"https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css\\" integrity=\\"sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm\\" crossOrigin=\\"anonymous\\" className=\\"jsx-235099215\\" />
  </Head>
  <div className=\\"jsx-235099215\\">
    <Header />
    <DummyComponent />
    ÃŽ
  </div>
  <JSXStyle id=\\"235099215\\">
    h
    t
    m
    l
    ,
    b
    o
    d
    y
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    w
    h
    i
    t
    e
    s
    m
    o
    k
    e
    ;
    }
  </JSXStyle>
</Fragment>"
`;
